solidity/exportBuildArtifact.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+# set script location as working directory
+cd "$(dirname "$0")"
+
+# Define the artifacts directory
+artifactsDir="./artifacts/build-info"
+# Define the output file
+outputFile="./buildArtifact.json"
+
+# log that we're in the script
+echo 'Finding and processing hardhat build artifact...'
+
+# Find the latest JSON build artifact
+jsonFiles=$(find "$artifactsDir" -type f -name "*.json" | sort | tail -n 1)
```

Comment:
sorting even by last modification might not be correct here, should probably remove `sort` altogether. if someone's having issues I think the modus operandi should be `yarn clean` before `yarn build`

given the scenario of 3 builds as follows:
build1 - use contracts v1
build2 - use contracts v2
build3 - use contracts v1

build3 does not generate any new artifacts because the hash of contracts would be the same as build1

in ci/npm publishing this all works fine because we do a clean build so there'll only be one

Reply:
maybe this is a very specific example I'm thinking of - i'll update to find the most recently modified file